server/src/domain/album/album.service.ts

```diff
@@ -181,6 +183,10 @@ export class AlbumService {
     await this.access.requirePermission(authUser, Permission.ALBUM_READ, id);
 
     const existingAssetIds = await this.albumRepository.getAssetIds(id, dto.ids);
+    const allowedAssetIds = new Set([
+      ...(await this.access.checkAccess(authUser, Permission.ALBUM_REMOVE_ASSET, existingAssetIds)),
+      ...(await this.access.checkAccess(authUser, Permission.ASSET_SHARE, existingAssetIds)),
+    ]);
```

Comment:
This works, it might be a bit cleaner if these were two separate sets, like `canRemove` or `canShare` and then it was used below like `const hasAccess = canRemove.has(id) || canShare.has(id)`, but overall this looks good too.

Reply:
Do you find this more readable, while we still maintain a single map lookup per id?

```javascript
const canRemove = await this.access.checkAccess(authUser, Permission.ALBUM_REMOVE_ASSET, existingAssetIds);
const canShare = await this.access.checkAccess(authUser, Permission.ASSET_SHARE, existingAssetIds);
const allowedAssetIds = new Set([...canRemove, ...canShare]);
```

Reply:
That looks better to me, yes. Thanks